internal/errors/errors.go

```diff
@@ -0,0 +1,96 @@
+// @license
+// Copyright 2025 Dynatrace LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	"fmt"
+)
+
+// ErrorClient represents error wrapper that wraps errors happening on the clients
+type ErrorClient struct {
+	// Wrapped contained the raw error returned from the client
+	Wrapped error
+
+	// Identifier being not empty the error message will contain the value set
+	Identifier string
+
+	// Operation is the HTTP method that was used by the client
+	Operation string
+
+	// Resource is a string of the resource
+	Resource string
+
+	// Reason can be provided why the error is happening
+	Reason string
+}
+
+func (e ErrorClient) Error() string {
+	if e.Identifier != "" {
+		return fmt.Sprintf("failed to %s %s with id %s: %v", e.Operation, e.Resource, e.Identifier, e.Wrapped)
+	}
+
+	return fmt.Sprintf("failed to %s %s: %v", e.Operation, e.Resource, e.Wrapped)
+}
+
+func (e ErrorClient) Unwrap() error {
+	return e.Wrapped
+}
+
+type ErrorValidation struct {
```

Comment:
We should add when what error should be used. `ErrorValidation` for example is meaningless. What validation? Server side? Client Side? Why set a field? Is this something that can be reacted to? Why is it not `ErrorClient`?